snitun/server/worker.py

```diff
@@ -110,16 +114,28 @@ def run(self) -> None:
 
             new[0].setblocking(False)
             asyncio.run_coroutine_threadsafe(
-                self._async_new_connection(*new), loop=self._loop,
+                self._async_new_connection(*new),
+                loop=self._loop,
             )
 
         # Shutdown worker
-        _LOGGER.info("Stop worker: %s", self.name)
+        _LOGGER.info("Stoping worker: %s", self.name)
+        asyncio.run_coroutine_threadsafe(
+            self._list_peer.stop(),
+            loop=self._loop,
+        ).result()
+        asyncio.run_coroutine_threadsafe(
+            self._list_sni.stop(),
+            loop=self._loop,
+        ).result()
```

Comment:
Let's first stop the SNI connection, so we have less todo at the peer shutdown as it's `one <-- many`

Reply:
These are not started, so they cant be stopped.